sdks/java/extensions/zetasketch/src/main/java/org/apache/beam/sdk/extensions/zetasketch/HllCountMergePartialFn.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.zetasketch;
+
+import com.google.zetasketch.HyperLogLogPlusPlus;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.coders.AtomicCoder;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderRegistry;
+import org.apache.beam.sdk.coders.NullableCoder;
+import org.apache.beam.sdk.extensions.zetasketch.HllCountMergePartialFn.HyperLogLogPlusPlusWrapper;
+import org.apache.beam.sdk.transforms.Combine;
+
+/**
+ * {@link Combine.CombineFn} for the {@link HllCount.MergePartial} combiner.
+ *
+ * @param <T> type of the HLL++ sketch to be merged
+ */
+class HllCountMergePartialFn<T>
+    extends Combine.CombineFn<byte[], HyperLogLogPlusPlusWrapper<T>, byte[]> {
+
+  // Call HllCountMergePartialFn.create() to instantiate
+  private HllCountMergePartialFn() {}
+
+  static HllCountMergePartialFn<?> create() {
+    return new HllCountMergePartialFn();
+  }
+
+  /**
+   * Accumulator for the {@link HllCount.MergePartial} combiner. Cannot use {@link
+   * HyperLogLogPlusPlus} directly because we need an "identity element" for the {@code merge}
+   * operation, and we are not able to create such "identity element" of type {@link
+   * HyperLogLogPlusPlus} without knowing the sketch precision and value type.
+   */
+  static final class HyperLogLogPlusPlusWrapper<T> {
```

Comment:
I don't think that will work. If we do that, then, for example, in our `addInput` function, the code will look like this:

```
  public HyperLogLogPlusPlus<T> addInput(
      HyperLogLogPlusPlus<T> accumulator, byte[] input) {
    if (accumulator == null) {
      // This will NOT work because accumulator is a local variable
      accumulator =  (HyperLogLogPlusPlus<T>) HyperLogLogPlusPlus.forProto(input);
    } else {
      accumulator.merge(input);
    }
    return accumulator;
  }
```

So we need the wrapper class as accumulator in order to be able to change its state.